File: frontend/src/types/emotion.ts
export interface EmotionVector {
  valence: number;   // -1.0 ~ 1.0
  arousal: number;   // -1.0 ~ 1.0
  dominance: number; // -1.0 ~ 1.0
}

export interface AnalyzeRequest {
  text: string;
}

export interface AnalyzeResponse {
  emotion: EmotionVector;
  seed_hex: string;  // "#RRGGBB" — backend provides base color only
  label: string;     // English slug, e.g. "anxious_relief"
}

export interface DerivedPalette {
  base: string;
  analogous: [string, string];
  complement: string;
  gradient: [string, string, string, string];
  textColor: string;
}

export interface EmotionState {
  current: AnalyzeResponse | null;
  previous: AnalyzeResponse | null;
  palette: DerivedPalette | null;
  isLoading: boolean;
  lerpProgress: number; // 0.0 ~ 1.0
  error: string | null;
}

File: backend/app/schemas/emotion.py
from pydantic import BaseModel, Field, field_validator
import re


class EmotionVector(BaseModel):
    valence: float = Field(..., ge=-1.0, le=1.0)
    arousal: float = Field(..., ge=-1.0, le=1.0)
    dominance: float = Field(..., ge=-1.0, le=1.0)


class AnalyzeRequest(BaseModel):
    text: str = Field(..., min_length=1, max_length=500)


class AnalyzeResponse(BaseModel):
    emotion: EmotionVector
    seed_hex: str = Field(..., pattern=r"^#[0-9A-Fa-f]{6}$")
    label: str  # English slug, e.g. "anxious_relief"

    @field_validator("seed_hex")
    @classmethod
    def validate_hex(cls, v: str) -> str:
        if not re.match(r"^#[0-9A-Fa-f]{6}$", v):
            raise ValueError("seed_hex must be a valid 6-digit hex color")
        return v.upper()
